node/self_connection.js

```diff
@@ -76,4 +77,12 @@ TChannelSelfConnection.prototype.buildOutResponse = function buildOutResponse(in
     }
 };
 
+TChannelSelfConnection.prototype.ping = function ping() {
+    var self = this;
+    var id = self.idCount++;
+    var body = new v2.PingResponse();
+    var res = new v2.Frame(id, body);
+    self.handlePingResponse(res);
```

Comment:
we can just get away with an anonymous object in both cases for now, with something like `// TODO: explicit type`:

So here we just have: 
```
self.pingResponseEvent.emit({self, 
  id: id
});
```

Above, in connection, we have:
```
self.pingResponseEvent.emit({self, 
  id: resFrame.id
});
```